src/browser/page-switch.js

```diff
@@ -31,21 +30,35 @@ define((require, exports, moudle) => {
         }
       }
     }, [
-      DOM.span({
+      DOM.div({
         key: 'thumbnail',
         className: 'tab-thumbnail',
-        style: {backgroundImage: readThumbnailURI(webViewerCursor.get('location'))},
-      }),
+      }, [
+        DOM.img({
+          key: 'image',
+          src: thumbnail,
+          alt: '',
+          onLoad: event => URL.revokeObjectURL(event.target.src)
```

Comment:
1. you store thumbnail in the state. That will break on restart, as the blobs are not stored across sessions
2. you revoke the URL after load, making re-using the blob impossible even in the same session